src/vivarium_census_prl_synth_pop/results_processing/addresses.py

```diff
@@ -18,6 +19,98 @@
 }
 
 
+def get_address_id_maps(
+    column_name: str,
+    obs_data: Dict[str, pd.DataFrame],
+    artifact: Artifact,
+    randomness: RandomnessStream,
+):
+    """
+    Get all maps that are indexed by `address_id`.
+
+    Parameters
+    ----------
+    column_name
+        Name of the column to use as an index
+    obs_data
+        Observer DataFrame with key for the observer name
+    artifact
+        A vivarium Artifact object needed by mapper
+    randomness
+        RandomnessStream to use in choosing zipcodes proportionally
+
+    Returns
+    -------
+    A dictionary of pd.Series suitable for pd.Series.map, indexed by `address_id`
+
+    """
+    if column_name != "address_id":
+        raise ValueError(f"Expected `address_id`, got `{column_name}`")
+    maps = dict()
+    maps.update(get_zip_map(column_name, obs_data, randomness))
+    maps.update(get_household_address_map(column_name, obs_data, artifact, randomness))
+    return maps
+
+
+def get_zip_map(
+    column_name: str,
+    obs_data: Dict[str, pd.DataFrame],
+    randomness: RandomnessStream,
+) -> Dict[str, pd.Series]:
+    """Gets a mapper for `address_id` to zipcode, based on state, puma, and proportion.
+
+    Parameters
+    ----------
+    column_name
+        Name of the column to use as an index
+    obs_data
+        Observer DataFrame with key for the observer name
+    randomness
+        RandomnessStream to use in choosing zipcodes proportionally
+
+    Returns
+    -------
+    A pd.Series suitable for pd.Series.map, indexed by column_name, with key "zipcode"
+
+    """
+    zip_map_dict = {}
+    output_cols = [column_name, "state", "puma"]  # columns in the output we use to map
+    address_ids = format_data_for_mapping(
```

Comment:
Calling `format_data_for_mapping` separately inseide each of these child functions is inefficient, as we are doing this concat operation multiple times. It'd be better to do it once inside `get_address_id_maps` with all the columns needed by any of the individual parts of the address.

Reply:
I'll make this change. I'll need to coordinate with @albrja to update his unmerged branch based on this branch.